File: spadea-android/libcommon/src/main/java/com/qpyy/libcommon/base/BaseDialog.java
package com.qpyy.libcommon.base;

import android.app.Dialog;
import android.content.Context;
import android.support.annotation.NonNull;

import com.qpyy.libcommon.R;

import butterknife.ButterKnife;

public abstract class BaseDialog extends Dialog {

    public BaseDialog(@NonNull Context context) {
        this(context, R.style.BaseDialogStyle);
    }

    public BaseDialog(@NonNull Context context, int themeResId) {
        super(context, themeResId);
        setContentView(getLayoutId());
        ButterKnife.bind(this);
        initView();
        initData();
    }


    public abstract int getLayoutId();

    public abstract void initView();

    public abstract void initData();


}

File: spadea-android/app/src/main/java/com/spadea/yuyin/base/BaseDialog.kt
package com.spadea.yuyin.base

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import com.spadea.yuyin.R

/**
 * Copyright (c) 1
 *
 * @Description
 * @Author         1
 * @Copyright      Copyright (c) 1
 * @Date           $date$ $time$
 */
abstract class BaseDialog(mContext: Context?, res: Int) : Dialog(mContext, res) {

    constructor(mContext: Context) : this(mContext, R.style.custom_dialog)

    abstract val layoutRes: Int

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layoutRes)
        initView()
    }

    private fun initView() {
        initLogic()
        setListener()
        processLogic()
    }


    protected abstract fun initLogic()
    protected abstract fun setListener()
    protected abstract fun processLogic()
}